garage/tf/core/networks.py

```diff
@@ -0,0 +1,48 @@
+import tensorflow as tf
+
+
+def mlp(input_var,
+        output_dim,
+        hidden_sizes,
+        hidden_nonlinearity=tf.nn.relu,
+        output_nonlinearity=None,
+        name="mlp",
+        hidden_w_init=tf.contrib.layers.xavier_initializer(),
+        hidden_b_init=tf.zeros_initializer(),
+        output_w_init=tf.contrib.layers.xavier_initializer(),
+        output_b_init=tf.zeros_initializer(),
+        batch_normalization=False,
+        weight_normalization=False,
+        is_training=True):
+
+    with tf.variable_scope(name):
```

Comment:
don't forget to include the input tensors in the name_scope call, as well as using their style of defining a default name (i.e. your function signature will have `name=None`)

Reply:
Why do we need name_scope here? Since we would always have a variable scope wrapping the mlp and both the variables and tensors would be already under the variable scope. Is there a case that we need to separate the operations and variables under different scope (when all of them belong to mlp)?

Also is it not a good practice if we put the input placeholder outside the scopes of mlp? In this way the input placeholder can be fed to different mlp to reduce redundancy (e.g. q_network and target q_network).

Reply:
* Using name scope will create a block in the TensorFlow graph which contains all ops created by your function call. This makes the graph much easier to read when viewed with TensorBoard
* Correct. A utility function for making networks (like this one) should never create a placeholder. It should only produce outputs from inputs. If you're suggesting that my comment implied you need to create a placeholder, I did not intend to say that.